mailet/standard/src/main/java/org/apache/james/transport/matchers/HasMimeType.java

```
@@ -0,0 +1,74 @@
+ ****************************************************************/
+
+
+package org.apache.james.transport.matchers;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import org.apache.james.core.MailAddress;
+import org.apache.mailet.Mail;
+import org.apache.mailet.base.GenericMatcher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.activation.MimeType;
+import javax.activation.MimeTypeParseException;
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.Set;
+
+
+/**
+ * <p>This matcher checks if the content type matches.</p>
+ *
+ * use: <pre><code><mailet match="HasMimeType=text/plain,text/html" class="..." /></code></pre>
+ */
+public class HasMimeType extends GenericMatcher {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(HasMimeType.class);
+
+    private Set<String> acceptedContentTypes;
+
+    public void init() throws javax.mail.MessagingException {
+        acceptedContentTypes = ImmutableSet.copyOf(Splitter.on(",").trimResults().split(getCondition()));
+    }
+
+    public Collection<MailAddress> match(Mail mail) throws javax.mail.MessagingException {
+        Optional<String> mimeTypes = getMimeTypeFromMessage(mail.getMessage());
+
+        return mimeTypes.filter(acceptedContentTypes::contains)
+                .map(any -> mail.getRecipients())
+                .orElse(ImmutableList.of());
+    }
+
+    private static Optional<String> getMimeTypeFromMessage(MimeMessage message) throws MessagingException {
+        try {
+            return Optional.of(new MimeType(message.getContentType()).getBaseType());
+        } catch (MimeTypeParseException e) {
+            LOGGER.warn(String.format("Error while parsing message's mimeType %s", message.getContentType()), e);
```

you don't really need String.format, you can use a slf4j placeholder for that

---

There's a running joke about swallowed exception here (SLF4J handle placeholders badly when combined with exception)

Better take no risk.